src/internal/interpreter/mod.rs

```diff
@@ -16,28 +21,29 @@ pub struct Interpreter {
     /// Holds temporary values
     stack: Vec<Constant>,
 
-    // Constant pool
-    constant_pool: Vec<Constant>,
+    code_holder: CodeHolder,
 
     /// Holds global variables
-    global: Vec<Option<Constant>>
+    global: Vec<Option<Constant>>,
 }
 
 impl Interpreter {
     /// Creates a new `Interpreter` instance
-    pub fn new() -> Interpreter {
+    pub fn new(ch: CodeHolder) -> Interpreter {
         Interpreter {
             accumulator: 0.0,
             call_stack: Vec::new(),
             stack: Vec::new(),
-            constant_pool: Vec::new(),
-            global: Vec::new()
+            code_holder: ch,
+            global: Vec::new(),
         }
     }
-}
 
-impl Default for Interpreter {
-    fn default() -> Self {
-        Self::new()
+    /// Reads a file at a given path, parses it, and creates an [`Interpreter`] instance.
+    ///
+    /// This is a convenience wrapper for [`crate::api::read_bytecode_file`] and behaves the same
+    /// way.
+    pub fn from_file(path: &str) -> Result<Interpreter, Error> {
```

Comment:
Separate this from `Default`, it simply doesn't make sense here.

Reply:
I don't have a better place to move it right now, so you can move it or let me know if you can find a better place

Reply:
something like this should be fine
```rust
impl Interpreter {
    pub fn from_file(path: &str) -> ...
}
```

Reply:
It has been updated